docs/DeveloperGuide.adoc

```diff
@@ -569,47 +569,58 @@ the module database up to date, or to cater to their own needs.
 [NOTE]
 The commands pertaining to module database administration can only be executed by users of type `Admin`.
 
+Administrators can choose to add, remove or edit a module using `addModuleDB`, `removeModuleDB` and `editModule` respectively.
+The changes will then be reflected
+in both the module list in memory and in storage. Figure 1(shown below) describes the general flow of the administration
+of the module database in an activity diagram.
+
+image::ADforModuleDatabaseAdministration.png[width="800]
+_Figure 1. Activity diagram for module database administration._
+
 ==== Current Implementation
 
 The Module database administration mechanism is facilitated by `ModuleList` in `Model` via the following operations:
 
 * `ModuleList#addModule(...)` -- This method adds a module to the module list.
 * `ModuleList#removeModule(...)` -- This method removes a module from the module list.
 * `ModuleList#hasModule(...)` -- This method checks if the given module already exists.
+* `ModuleList#updateModule(...)` -- This method replace the specified target module with the given edited module.
 
 [NOTE]
 These operations are exposed in the `Model` interface as `Model#addModuleToDatabase(...)`,
-`Model#removeModuleFromDatabase(...)` and `Model#hasModuleInDatabase(...)` respectively.
-
+`Model#removeModuleFromDatabase(...)`, `Model#hasModuleInDatabase(...)` and `Model#updateModule(...)` respectively.
 
 Module database administration mechanism is facilitated by several command classes in modsUni.
 
 **Adding module to database**
 
 `addModuleDB` mechanism is facilitated by `AddModuleToDatabaseCommand`. It allows the addition of modules to database.
```

Comment:
I think its a good idea to give an overview of what command classes.
Maybe it could be in a point form like:
```
Module database administration mechanism is facilitated by several command classes in modsUni.
- addModuleDB
- removeModuleDB
...
```

Reply:
List added.